use new MappingMonitor

arnej27959 opened the pull request:
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@havardpe please review

arnej27959 committed:
use new MappingMonitor

havardpe commented on slobrok/src/vespa/slobrok/server/local_rpc_monitor_map.cpp:
```diff
@@ -29,28 +25,34 @@ void LocalRpcMonitorMap::DelayedTasks::PerformTask() {
 LocalRpcMonitorMap::PerService *
-LocalRpcMonitorMap::lookup(ManagedRpcServer *rpcsrv) {
-    auto iter = _map.find(rpcsrv->getName());
+LocalRpcMonitorMap::lookup(const ServiceMapping &mapping) {
```
this seems non-optional again, consider simplifying usage.

havardpe commented on slobrok/src/vespa/slobrok/server/local_rpc_monitor_map.cpp:
```diff
@@ -97,95 +100,94 @@ void LocalRpcMonitorMap::doAdd(const ServiceMapping &mapping) {
             return;
         }
         PerService removed = std::move(exists);
+        ServiceMapping toRemove{old->first, removed.spec};
```
consider using a helper function that removes something from the map, calls stop and returns a PerService

havardpe commented on slobrok/src/vespa/slobrok/server/local_rpc_monitor_map.cpp:
```diff
@@ -97,95 +100,94 @@ void LocalRpcMonitorMap::doAdd(const ServiceMapping &mapping) {
     auto [ iter, was_inserted ] =
         _map.try_emplace(mapping.name, globalService(mapping));
     LOG_ASSERT(was_inserted);
+    _mappingMonitor->start(mapping);
```
consider using a helper function to add a PerService and call start

arnej27959 committed:
now lookup never fails

arnej27959 committed:
refactor common code -> addToMap/removeFromMap

* note: extra RemovedData struct is mainly useful
  to avoid warnings from -Winline (pragma was not
  completely successful in suppressing these).
